server/src/main/java/io/deephaven/server/table/stats/CharacterChunkedStats.java

```diff
@@ -35,76 +32,67 @@ public Table processChunks(final RowSet rowSet, final ColumnSource<?> columnSour
         long count = 0;
         int uniqueCount = 0;
 
-        final TObjectLongHashMap<Object> countValues = new TObjectLongHashMap<>();
+        final TCharLongHashMap countValues = new TCharLongHashMap();
         boolean useSet = false;
-        final Set<Object> uniqueValues = new HashSet<>();
-
-        try (final ChunkSource.GetContext getContext = columnSource.makeGetContext(CHUNK_SIZE)) {
-            final RowSequence.Iterator rsIt = rowSet.getRowSequenceIterator();
-
-            while (rsIt.hasMore()) {
-                // Grab up to the next CHUNK_SIZE rows
-                final RowSequence nextKeys = rsIt.getNextRowSequenceWithLength(CHUNK_SIZE);
-
-                final CharChunk<? extends Values> chunk =
-                        (usePrev ? columnSource.getPrevChunk(getContext, nextKeys)
-                                : columnSource.getChunk(getContext, nextKeys)).asCharChunk();
-                final int chunkSize = chunk.size();
-                for (int ii = 0; ii < chunkSize; ii++) {
-                    final char val = chunk.get(ii);
-
-                    if (val == QueryConstants.NULL_CHAR) {
-                        continue;
-                    }
-
-                    count++;
-
-                    if (useSet) {
-                        uniqueValues.add(val);
-                    } else if (uniqueCount > maxUniqueToCollect) {
-                        // we no longer need to track counts for these items; fall back to a Set to get at least a count
-                        uniqueValues.addAll(countValues.keySet());
-                        countValues.clear();
-                        uniqueValues.add(val);
-                        useSet = true;
-                    } else if (countValues.adjustOrPutValue(val, 1, 1) == 1) {
-                        uniqueCount++;
-                    }
+        final TCharSet uniqueValues = new TCharHashSet();
+
+        try (CharacterColumnIterator iterator =
+                new ChunkedCharacterColumnIterator(usePrev ? columnSource.getPrevSource() : columnSource, rowSet)) {
+            while (iterator.hasNext()) {
+                char val = iterator.next();
```

Comment:
This is important to change, or you will NPE when `NULL_CHAR` is converted to `null` and then unboxed. I wonder if this means the null case has not been tested?
```suggestion
                char val = iterator.nextChar();
```

Reply:
Good catch from the col iterator change. I've added a simple test for the remaining types to check these.